src/parser/mod.rs

```diff
@@ -864,14 +846,7 @@ impl Parser {
                         self.next_expected_keyword(Keyword::On)?;
                         self.next_token();
 
-                        Some(match self.parse_expression(Precedence::Lowest)? {
-                            Some(expr) => expr,
-                            None => {
-                                return Err(Error::ParseErr(fmt_err!(
-                                    "ON Predicate expression is not valid!"
-                                )));
-                            }
-                        })
+                        Some(self.parse_expression(Precedence::Lowest)?)
```

Comment:
I think we should keep this error message here.

Reply:
ok,Pushed

Reply:
> I think we should keep this error message here.

And in other place. you hide the error message. May should keep them too.

Reply:
ok

Reply:
> ok

Pushed @fansehep